podcasts/Analytics/WidgetAnalytics.swift

```diff
@@ -0,0 +1,48 @@
+import Foundation
+import WidgetKit
+
+class WidgetAnalytics {
+    private let userDefaults: UserDefaults
+    private let analytics: Analytics
+
+    init(userDefaults: UserDefaults = UserDefaults.standard,
+         analytics: Analytics = .shared) {
+        self.userDefaults = userDefaults
+        self.analytics = analytics
+    }
+
+    func track() {
+        var previousWidgets = userDefaults.dictionary(forKey: "installed-widgets") as? [String: Bool] ?? [String: Bool]()
+        var currentWidgets: [String] = []
+
+        WidgetCenter.shared.getCurrentConfigurations { [self] widgetInfos in
```

Comment:
I planned to add unit tests for this logic, but I'm not able to mock `WidgetInfo`, so it would be a lot of additional code, structs, and protocols to achieve this.